test/e2e/service.go

```diff
@@ -180,4 +181,131 @@ var _ = Describe("Services", func() {
 		Expect(foundRW).To(Equal(true))
 		Expect(foundRO).To(Equal(true))
 	})
+
+	It("should serve basic a endpoint from pods", func(done Done) {
+		serviceName := "endpoint-test"
+		ns := api.NamespaceDefault
+		labels := map[string]string{
+			"foo": "bar",
+			"baz": "blah",
+		}
+
+		defer func() {
+			err := c.Services(ns).Delete(serviceName)
```

Comment:
Yeah, @roberthbailey and I just chatted and that was my understanding as well. I believe the defer's run at the tail of the main goroutine here. We should test this case eventually, though ...